# SPDX-License-Identifier: AGPL-3.0-or-later
# This file is part of yagips
# Â©2023 Alex Pensinger (ArcticLuma113)
# This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
# You should have received a copy of the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.

syntax = "proto3";

package proto;

message RegionSimpleInfo {
    string name = 1;
    string title = 2;
    string type = 3;
    string dispatch_url = 4;
}

message QueryRegionListHttpRsp {
    int32 retcode = 1;
    repeated RegionSimpleInfo region_list = 2;
    bytes client_secret_key = 5;
    bytes client_custom_config_encrypted = 6;
    bool enable_login_pc = 7;
}

message ForceUpdateInfo {
    string force_update_url = 1;
}

message StopServerInfo {
    uint32 stop_begin_time = 1;
    uint32 stop_end_time = 2;
    string url = 3;
    string content_msg = 4;
}

message ResVersionConfig {
    uint32 version = 1;
    bool relogin = 2;
    string md5 = 3;
    string release_total_size = 4;
    string version_suffix = 5;
    string branch = 6;
    string next_script_version = 7;
}

message RegionInfo {
    string gateserver_ip = 1;
    uint32 gateserver_port = 2;
    string pay_callback_url = 3;
    string area_type = 7;
    string resource_url = 8;
    string data_url = 9;
    string feedback_url = 10;
    string bulletin_url = 11;
    string resource_url_bak = 12;
    string data_url_bak = 13;
    uint32 client_data_version = 14;
    string handbook_url = 16;
    uint32 client_silence_data_version = 18;
    string client_data_md5 = 19;
    string client_silence_data_md5 = 20;
    ResVersionConfig res_version_config = 22;
    bytes secret_key = 23;
    string official_community_url = 24;
    string client_version_suffix = 26;
    string client_silence_version_suffix = 27;
    bool use_gateserver_domain_name = 28;
    string gateserver_domain_name = 29;
    string user_center_url = 30;
    string account_bind_url = 31;
    string cdkey_url = 32;
    string privacy_policy_url = 33;
    string next_resource_url = 34;
    ResVersionConfig next_res_version_config = 35;
    string game_biz = 36;
}

message QueryCurrRegionHttpRsp {
    int32 retcode = 1;
    string msg = 2;
    RegionInfo region_info = 3;
    bytes client_secret_key = 11;
    bytes region_custom_config_encrypted = 12;
    bytes client_region_custom_config_encrypted = 13;
    
    oneof detail {
        ForceUpdateInfo force_udpate = 4;
        StopServerInfo stop_server = 5;
    }
}
